pauthabielf64/pauthabielf64.rst

```diff
@@ -428,101 +435,66 @@ memory management unit. Platforms may not need to implement all of
-  // Authenticate to get unsigned pointer
-  autia x0, x8
-
-In the example the :got_auth: and :got_auth_lo12: operators result in
-AUTH variant GOT generating relocations being used.
-
-Compatibility
-^^^^^^^^^^^^^
-
-If there are AUTH and non-AUTH variant GOT generating relocations to
-the same symbol two GOT entries are required, one signed and one
-unsigned. While not a hard limitation many static linkers only support
-a single GOT entry per symbol. An implementation may choose to fault
-an AUTH and a non-AUTH GOT generating relocation to the same symbol,
-this would require all the GOT-generating relocations to a symbol to
-be signed or unsigned.
+created as a result of a GOT-creating relocation from a relocatable
+object. The GOT is normally split into two subsets, the GOT and the
+PLT GOT. With the PLT GOT made up of code pointers that are only
+accessed by linker generated PLT sequences. As the PLT GOT is not
+accessed directly by code in relocatable objects will be covered in a
+separate section of the document.
+
+The pointers in the GOT may include pointers to code so there is a
+question of whether these pointers should be signed and if so, how
+should they be signed?
+
+The value of many of the pointers in the GOT is not known at static
+link time, these GOT entries will have a dynamic relocation that a
+dynamic linker will resolve at program load time. Once the GOT has
+been relocated it can be re-mapped as read-only (RELRO). If the GOT is
+RELRO then the GOT does not need to be signed.
+
+As a majority of platforms support RELRO and assuming a RELRO GOT
+simplifies the ABI, this document will assume an unsigned GOT. An
+optional appendix describes how a GOT can be signed.
 
 PLT GOT signing
 ---------------
 
 The PLT is a table of trampolines used to indirect function calls
 through a function pointer. The PLT GOT is a subset of the GOT that is
-used exclusively by the PLT. When lazy binding is used the initial
-contents of the PLT GOT point to the first entry in the PLT which is
-reserved for the lazy resolver. When a function is called via its PLT
-entry control is transferred to the lazy resolver which finds the
-destination address and writes it back to the PLT GOT. As the lazy
-resolver needs to write to PLT GOT, it requires the PLT GOT to be
-writeable throughout the lifetime of the program. When lazy binding is
-disabled all relocations are resolved at load time and the PLT GOT can
-be made read-only after relocation like the GOT.
-
-If the PLT GOT is signed the dynamic linker must sign
-all entries in the table during program loading and the static linker must
-generate PLT trampolines that authenticate pointers loaded from the PLT GOT.
+used exclusively by the PLT.
+
+When lazy binding is disabled all relocations are resolved at load
+time and the PLT GOT can be made RELRO like the GOT. In this case the
+PLT GOT does not need to be signed.
+
+When lazy binding is enabled the initial contents of the PLT GOT point
```

Comment:
comma?: .. is enabled`,` the initial..

point -> points

Reply:
I've reworded to say 
"When lazy binding is enabled the initial contents of each entry in the
PLT GOT points to the first entry in the PLT. The first entry in the PLT is
reserved for the lazy resolver."